libs/sdk-bindings/bindings-android/lib/src/main/kotlin/breez_sdk_notification/ForegroundService.kt

```diff
@@ -0,0 +1,152 @@
+        stopSelf()
+    }
+
+    // =========================================================== //
+    //                    START COMMAND HANDLER                    //
+    // =========================================================== //
+
+    /** Called when an intent is called for this service. */
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+        val intentDetails = "[ intent=$intent, flag=$flags, startId=$startId ]"
+        Logger.tag(TAG).debug { "Start foreground service from intent $intentDetails" }
+
+        // Connect to SDK if source intent has data message with valid payload
+        getConnectRequest()?.let { connectRequest ->
+            getJobFromIntent(intent)?.also { job ->
+                launchSdkConnection(connectRequest, job)
+            } ?: run {
+                Logger.tag(TAG).warn { "Received invalid data message." }
+                shutdown()
+            }
+        } ?: run {
+            Logger.tag(TAG).warn { "Missing ConnectRequest." }
+            shutdown()
+        }
+
+
+        return START_NOT_STICKY
+    }
+
+    /** To be implemented by the application foreground service.
+     *  It should retrieve the Breez API key and node mnemonic then construct
+     *  a ConnectRequest to be used to call the Breez SDK connect function. */
+    abstract fun getConnectRequest(): ConnectRequest?
+
+    /** Get the job to be executed from the Message data in the Intent.
+     *  This can be overridden to handle custom jobs. */
+    open fun getJobFromIntent(intent: Intent?): Job? {
+        return getMessage(intent)?.let { message ->
+            message.payload?.let { payload ->
+                when (message.type) {
+                    Constants.MESSAGE_TYPE_PAYMENT_RECEIVED -> ReceivePaymentJob(
+                        applicationContext,
+                        this,
+                        payload
+                    )
+
+                    Constants.MESSAGE_TYPE_LNURL_PAY_INFO -> LnurlPayInfoJob(
+                        applicationContext,
+                        this,
+                        payload
+                    )
+
+                    Constants.MESSAGE_TYPE_LNURL_PAY_INVOICE -> LnurlPayInvoiceJob(
+                        applicationContext,
+                        this,
+                        payload
+                    )
+
+                    else -> null
+                }
+            }
+        }
+    }
+
+    @Suppress("DEPRECATION")
+    /** Get the Message parcel from the Intent. */
+    open fun getMessage(intent: Intent?): Message? {
+        return intent?.let {
+            return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) it.getParcelableExtra(
+                Constants.EXTRA_REMOTE_MESSAGE,
+                Message::class.java
+            ) else it.getParcelableExtra(Constants.EXTRA_REMOTE_MESSAGE)
+        }
+    }
```

Comment:
Can this be part of `Message.kt`?

Reply:
Done [d6939d9](https://github.com/breez/breez-sdk/pull/803/commits/d6939d9246c582e50327b988a1ce75236c4f8b9f)